features/notifications.yml

```diff
@@ -1,38 +1,52 @@
 name: Notifications
-description: The notifications API sends system notifications to the user, often even when the page or browser is not the foreground application.
-spec: https://notifications.spec.whatwg.org/
-caniuse: notifications
-# Notifications support is complicated and this entry is unusual. It represents
-# some kind of notification support available to web pages. More fine-grained
-# features could be added to represent specific things that work.
-#
-# https://caniuse.com/notifications marks iOS support as partial with this note:
-# "Requires website to first be added to the Home Screen."
+# This feature is the general Notifications API, and does not anchor to the
+# Caniuse feature.
 #
-# BCD doesn't use partial_implementation for safari_ios, but has this note:
-# "Notifications are supported in web apps saved to the home screen."
+# See `features/notifications-pwa.yml` for the Notifications feature for mobile
+# browsers that have platform-specific requirements and aligns with Caniuse.
+description: The Notifications API displays messages to the user outside of the browser, typically using the notification system of the host operating system. This feature is solely client-side, and does not integrate with server-originating notifications.
+spec: https://notifications.spec.whatwg.org/
+# Computing from the ServiceWorker feature since that gets us a baseline
+# which matches the general availability of the feature given platform
+# constraints.
 #
-# We do not have the concept of partial support or notes:
-# https://github.com/web-platform-dx/web-features/issues/915
+# This matches Caniuse: https://caniuse.com/?search=notifications
 #
-# Another complication is the two entry points to this API:
-# - new Notification() for non-persistent notifications.
-# - showNotification() in service workers for persistent notifications.
+# BCD notes:
 #
-# Finally, BCD and caniuse.com disagree about whether it was Safari 6 or 7 that
-# first supported notifications, and BCD originally agreed with caniuse.com:
-# https://github.com/mdn/browser-compat-data/pull/7949
+# https://github.com/mdn/browser-compat-data/blob/0398a9fa847b3a616fa7968f79aadfc6cba4718f/api/Notification.json#L18
 #
-# Due to this complexity, manually combine these 3 data points:
-# https://developer.mozilla.org/en-US/docs/Web/API/Notification/Notification#browser_compatibility
-# https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/showNotification#browser_compatibility
-# https://caniuse.com/notifications (excluding partial support on iOS)
-status:
-  baseline: false
-  support:
-    chrome: "20"
-    chrome_android: "42"
-    edge: "14"
-    firefox: "22"
-    firefox_android: "22"
-    safari: "6"
+compat_features:
```

Comment:
One problem we now have with this is that there isn't a BCD key that we can use to compute the browser support. There isn't a BCD key that says "supported in the main thread of regular websites". This means that the status now depends on all of the keys below, which have widely different browser support, because they represent later additions to the spec.
Should we add a new BCD key to simplify things?

Reply:
I'm not familiar with how BCD keys are birthed. Is that a typical approach to this kind of conundrum?

Reply:
BCD keys usually represent properties, methods, events, interfaces, etc. So, things that you can actually use as a developer.
But, there's also a small number of BCD keys that represent "special cases". Like when a method can be used with a new type of parameter which couldn't be used before.
When a feature evolves in a substantial new way, even if this evolution doesn't come with a new method or property, we tend to create a new BCD key to track this evolution across browsers.

In the present case, I would suggest creating a new BCD key in [Notification.json](https://github.com/mdn/browser-compat-data/blob/main/api/Notification.json) like `api.Notification.website_main_thread_support`.

For Chrome Android, this key would have the following support data:

```json
"chrome_android": {
  "version_added": false,
  "notes": "On Chrome for Android, notifications are only available through service workers."
}
```

And for Safari iOS:

```json
"safari_ios": {
  "version_added": false,
  "notes": "On Safari for iOS, notifications are only supported in web apps saved to the home screen."
}
```

I know that BCD already has notes for these caveats, but there's no BCD key that you can look at to know exactly which browsers support sending notifications from a webpage's main thread.

I'll create a PR in BCD in a minute that get that discussion going.

Reply:
See https://github.com/mdn/browser-compat-data/pull/24705#issuecomment-2425873063.